src/main/java/org/spongepowered/api/text/Text.java

```diff
@@ -107,6 +130,8 @@
         this.clickAction = Optional.<ClickAction<?>>ofNullable(clickAction);
         this.hoverAction = Optional.<HoverAction<?>>ofNullable(hoverAction);
         this.shiftClickAction = Optional.<ShiftClickAction<?>>ofNullable(shiftClickAction);
+        this.childrenIterable = () ->
+                this.children.isEmpty() ? Iterators.singletonIterator(this) : new TextIterator(this);
```

Comment:
In the first example the values is set outside the constructor.
Usually if something is set within the ctor I assume it is dependent on the ctor arguments, but `childrenIterable` isn't.